logstash-core/lib/logstash/environment.rb

```diff
@@ -45,6 +45,7 @@ module Environment
            Setting::Boolean.new("pipeline.reloadable", true),
            Setting::Boolean.new("pipeline.plugin_classloaders", false),
            Setting::Boolean.new("pipeline.separate_logs", false),
+           Setting::Boolean.new("pipeline.ordered", true),
```

Comment:
I assumed that since the Ruby execution behaviour when using a single worker was to preserve ordering, the default should be `true` so that if `pipeline.workers` is set to 1 it would behave the same was as the Ruby execution. We do actually consider the loss of ordering when using a sigle worker with Java execution a bug. I think that in v7 we should set it to `true` by default and in v8 put it back to false by default.